pennylane/transforms/split_non_commuting.py

```diff
@@ -168,6 +168,13 @@ def circuit0(x):
 
         def reorder_fn(res):
             """re-order the output to the original shape and order"""
+            if qml.active_return():
+                for i, tape in enumerate(tapes):
+                    if len(tape.observables) == 1:
```

Comment:
Minor - I have a vague recollection that we'd like to prefer the `measurements` attribute (could be wrong though)
```suggestion
                    if len(tape.measurements) == 1:
```